src/main/java/edu/pav/PatientTrackerSystem/controller/DoctorController.java

```diff
@@ -63,4 +72,46 @@ private List<Doctor> filterDoctorByStatus(List<Doctor> inputDoctors, boolean app
                 .filter(doctor -> doctor.getIsApproved() == approvalStatus)
                 .collect(Collectors.toList());
     }
+
+    private Boolean existsDoctor(String email) {
+        return doctorRepository.findByEmail(email) != null;
+    }
+
+    @Transactional
+    @PostMapping(value = "/doctors/signup")
+    public BaseResponse<String> signup(@RequestBody DoctorSignupRequest request) {
+
+        if(existsDoctor(request.getEmail())) {
+            return BaseResponse.<String>builder()
+                    .status(HttpStatus.BAD_REQUEST)
```

Comment:
Nitpick: Better error code might be 409 https://stackoverflow.com/questions/3825990/http-response-code-for-post-when-resource-already-exists

Reply:
Done